### PR TITLE
[WIP] Basic permissions

### PR DESCRIPTION
**Pull request by speza**

### CODE REVIEW

File: handlers/handler.go [hunk 1]
```diff
@@ -1,23 +1,15 @@
 package handlers
 
 import (
-	"crypto/rsa"
 	"errors"
-	"fmt"
-	"net/http"
-	"strings"
-
-	rice "github.com/GeertJohan/go.rice"
```
Comment (michelvocks) [inline handlers/handler.go]: Do you use the `go import` tool? https://godoc.org/golang.org/x/tools/cmd/goimports
This automatically formats the order of imports and also adds/removes imports. Actually the defacto standard.

File: roles.go [hunk 2]
```diff
@@ -0,0 +1,183 @@
+package gaia
+
+var (
+	// TODO: Load these in via a file
```
Comment (michelvocks) [inline roles.go]: Before we merge this PR we should do this or at least move this into the correct subpackage. The first directory level is not the best place for this.

File: gaia.go [hunk 3]
```diff
@@ -115,6 +114,53 @@ type User struct {
...
+}
+
+// UserRoleEndpoint represents the endpoint & method of the API that should be secured. Is most commonly used by
+// the API middleware to validate permission security.
+type UserRoleEndpoint struct {
+	Path   string `json:"path"`
+	Method string `json:"method"`
+}
+
+func NewUserRoleEndpoint(method string, path string) *UserRoleEndpoint {
```
[the remark this answers is not in the document]
Comment (speza) [reply gaia.go]: Noted!

File: frontend/client/views/settings/index.vue [hunk 4]
```diff
@@ -62,9 +66,9 @@
                     :rows="pipelineRows"
                     :paginate="true"
                     :global-search="true"
-                    :defaultSortBy="{field: 'id', type: 'desc'}"
+                    :defaultSortBy="{field: 'id', type: 'desc'}"Name
```
[the remark this answers is not in the document]
Comment (speza) [reply frontend/client/views/settings/index.vue]: Definitely not ha!

File: frontend/client/views/settings/index.vue [hunk 5]
```diff
@@ -31,8 +31,12 @@
                         <span>{{ convertTime(props.row.lastlogin) }}</span>
                       </td>
                       <td>
-                        <a v-on:click="editUserModal(props.row)"><i class="fa fa-edit" style="color: whitesmoke;"></i></a>
-                        <a v-on:click="deleteUserModal(props.row)" v-if="props.row.username !== session.username"><i class="fa fa-trash" style="color: whitesmoke;"></i></a>
+                        <a v-on:click="editUserModal(props.row)"><i class="fa fa-edit"
+                                                                    style="color: whitesmoke;"></i></a>
+                        <a v-on:click="userPermissionsModal(props.row)"><i class="fa fa-user"
```
Comment (michelvocks) [inline frontend/client/views/settings/index.vue]: It seems like this modal doesn't exist anymore?

**Comment by michelvocks** [comment]
Thanks @speza 
Looks and works really well! I got one last point (see review above) and then it should be ready for merge in my opinion. 😄

File: frontend/client/views/settings/index.vue (hunk 5 above)
Comment (speza) [reply frontend/client/views/settings/index.vue]: Very good point mate! Thanks for pointing it out

**Comment by speza** [comment]
@michelvocks removed old js perms stuff and pushed :+1:

**Pull request merged**